### PR TITLE
Exclude Stride check for 0 sized arrays

### PR DESCRIPTION
**Pull request by mitkotak**
Excluded stride check for 0 size arrays in `arraycontext.container.traversal.py`

### CODE REVIEW

File: arraycontext/container/traversal.py [hunk 1]
```diff
@@ -741,7 +741,10 @@ def _unflatten(template_subary: ArrayOrContainerT) -> ArrayOrContainerT:
             # {{{ check strides
 
             if strict and hasattr(template_subary, "strides"):
-                if template_subary.strides != subary.strides:
+                if (
+                    (template_subary.strides != subary.strides)
+                    and (template_subary.size != 0)
```
Comment (kaushikcfd) [inline arraycontext/container/traversal.py:747]: Please add a comment explaining why checking strides on 0-size arrays is ill-defined.

**Review by kaushikcfd — APPROVED** [review]
Thanks! LGTM, once the comment has been addressed. In the future, please keep a cleaner commit history, it's good practice and helps with the review process. Take a look at the docs of `git-rebase`, `git-reset`/

File: arraycontext/container/traversal.py (hunk 1 above)
Comment (mitkotak) [reply arraycontext/container/traversal.py:747]: Added comment

**Comment by kaushikcfd** [comment]
Reason for closing: Targeted to upstream in https://github.com/inducer/arraycontext/pull/175.

**Pull request closed without merging**